pkg/common/azclient/trafficmanagerclient/custom.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+// Package trafficmanagerclient provides a client for Azure Traffic Manager profiles.
+// It wraps the Azure SDK for Go's Traffic Manager client and provides additional functionality
+// such as metrics collection.
+package trafficmanagerclient
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/trafficmanager/armtrafficmanager"
+	"sigs.k8s.io/cloud-provider-azure/pkg/azclient/metrics"
+)
+
+const (
+	profileResourceType = "TrafficManagerProfile"
+)
+
+// ProfilesClient is a wrapper around the Azure Traffic Manager Profiles client.
+// It provides methods to interact with Traffic Manager profiles in Azure.
+type ProfilesClient struct {
+	*armtrafficmanager.ProfilesClient
+	subscriptionID string
+}
+
+// NewProfilesClient creates an Azure traffic manager profile client.
+func NewProfilesClient(client *armtrafficmanager.ProfilesClient, subscriptionID string) *ProfilesClient {
+	return &ProfilesClient{
+		ProfilesClient: client,
+		subscriptionID: subscriptionID,
+	}
+}
+
+// Get gets a Traffic Manager profile by its name and resource group.
+func (client *ProfilesClient) Get(ctx context.Context, resourceGroupName, profileName string, options *armtrafficmanager.ProfilesClientGetOptions) (resp armtrafficmanager.ProfilesClientGetResponse, err error) {
+	metricsCtx := metrics.BeginARMRequest(client.subscriptionID, resourceGroupName, profileResourceType, "get")
```

Comment:
this is a built in function?

Reply:
reused the metrics package in cloud provider repo, https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/pkg/azclient/metrics/metrics.go

Reply:
is there any advantage of this over our metrics? IIRC, we need to use telemetray provider in order to use this? What's the benefit of that overhead?